File: validation/poc-00/java-candidate/src/main/java/dev/vra/poc00/application/InventoryRepository.java
package dev.vra.poc00.application;

import dev.vra.poc00.domain.InventoryBalance;
import dev.vra.poc00.domain.SkuId;
import java.util.Optional;

public interface InventoryRepository {
    Optional<InventoryBalance> reserve(SkuId skuId, int quantity);
    boolean exists(SkuId skuId);
}

File: validation/poc-00/kotlin-candidate/src/main/kotlin/dev/vra/poc00/kotlin/application/InventoryRepository.kt
package dev.vra.poc00.kotlin.application

import dev.vra.poc00.kotlin.domain.InventoryBalance
import dev.vra.poc00.kotlin.domain.SkuId

interface InventoryRepository {
    fun reserve(skuId: SkuId, quantity: Int): InventoryBalance?
    fun exists(skuId: SkuId): Boolean
}
